docs/src/userguide/tempita.rst

```diff
@@ -0,0 +1,295 @@
+----------------------------------
+
+Conditional execution based on an expression.
+
+**Syntax**::
+
+    {{if condition}}
+      ... content if true ...
+    {{elif other_condition}}
+      ... content if elif is true ...
+    {{else}}
+      ... content if all above are false ...
+    {{endif}}
+
+**Example**:
+
+.. code-block:: python
+
+    >>> tmpl = Template("""
+    ... {{if x > 0}}
+    ... Positive
+    ... {{elif x == 0}}
+    ... Zero
+    ... {{else}}
+    ... Negative
+    ... {{endif}}
+    ... """)
+    >>> tmpl.substitute(x=-5)
+    'Negative\n'
+
+``for`` block
+-------------
+
+Iterate over a sequence and render the body for each element.
+
+**Syntax**::
+
+    {{for var in iterable}}
+      ... body using {{var}} ...
+    {{endfor}}
+
+You may also unpack tuples or iterate with multiple variables.
+
+**Example**:
+
+.. code-block:: python
+
+    >>> tmpl = Template("""
+    ... {{for name, score in scores}}
+    ... {{name}}: {{score}}
+    ... {{endfor}}
+    ... """)
+    >>> print(tmpl.substitute(scores=[("Alice", 95), ("Bob", 88)]))
+    Alice: 95
+    Bob: 88
+
+
+.. note:: Tempita ``for`` block supports also ``{{break}}`` and ``{{continue}}`` statements.
+
+Nesting Blocks
+--------------
+
+Blocks can be nested arbitrarily.
+
+.. code-block:: python
+
+    >>> tmpl = Template("""
+    ... {{for item in items}}
+    ...   {{if item < 0}}
+    ...   {{continue}}
+    ...   {{elif item % 2 == 0}}
+    ...   {{item}} is even
+    ...   {{else}}
+    ...   {{item}} is odd
+    ...   {{endif}}
+    ... {{endfor}}
+    ... """)
+    >>> print(tmpl.substitute(items=[-1, 1, 2, 3]))
+      1 is odd
+      2 is even
+      3 is odd
+
+Indentation, Whitespace, and Newlines
+-------------------------------------
+
+- The template preserves whitespace exactly as written around directives.
+- Newlines in the template become newlines in output.
+- Indentation is preserved, so control block bodies should be indented
+  meaningfully by the template author if you want nice output.
+
+Examples
+--------
+
+Here are consolidated examples showing most syntax in use:
+
+.. code-block:: python
+
+    from Cython.Tempita import Template
+
+    tmpl = Template("""
+    Header
+    {{# This is a comment }}
+
+    {{if user["is_admin"]}}
+      Welcome, Admin {{user["name"]}}!
+    {{else}}
+      Hello, {{user["name"] or 'Guest'}}.
+    {{endif}}
+
+    {{for item in items}}
+      * {{item["name"]}}: {{item["value"]}}
+    {{endfor}}
+
+    {{py: x = 1 + 2}}
+    Inline code result: {{x}}
+
+    Expression: 1 + 2 = {{1 + 2}}
+
+    """)
+
+    print(tmpl.substitute(user={'name': 'Bob', 'is_admin': False},
+                          items=[{'name':'A','value':10},
+                                 {'name':'B','value':20}]))
+
+Output::
+
+    Header
+      Hello, Bob.
+
+      * A: 10
+      * B: 20
+
+    Expression: 1 + 2 = 3
+
+Notes and Limitations
+---------------------
+
+- Templates are evaluated at runtime; errors show up when calling
```

Comment:
> runtime

This might be confused with "when running your Cython module". Maybe

> Templates are evaluated when they are substituted, rather than when they are created.